tests/scripts/thread-cert/Cert_9_2_02_MGMTCommissionerSet.py

```diff
@@ -0,0 +1,183 @@
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+from ipaddress import ip_address
+import unittest
+
+from mle import NetworkData
+from network_data import CommissioningData
+import command
+import config
+import mesh_cop
+import mle
+import node
+
+COMMISSIONER = 1
+LEADER = 2
+
+
+class Cert_9_2_02_MGMTCommissionerSet(unittest.TestCase):
+
+    def setUp(self):
+        self.simulator = config.create_default_simulator()
+
+        self.nodes = {}
+        for i in range(1,3):
+            self.nodes[i] = node.Node(i, simulator=self.simulator)
+
+        self.nodes[COMMISSIONER].set_panid(0xface)
+        self.nodes[COMMISSIONER].set_mode('rsdn')
+        self.nodes[COMMISSIONER].add_whitelist(self.nodes[LEADER].get_addr64())
+        self.nodes[COMMISSIONER].enable_whitelist()
+        self.nodes[COMMISSIONER].set_router_selection_jitter(1)
+
+        self.nodes[LEADER].set_panid(0xface)
+        self.nodes[LEADER].set_mode('rsdn')
+        self.nodes[LEADER].add_whitelist(self.nodes[COMMISSIONER].get_addr64())
+        self.nodes[LEADER].enable_whitelist()
+        self.nodes[LEADER].set_router_selection_jitter(1)
+
+    def tearDown(self):
+        for node in list(self.nodes.values()):
+            node.stop()
+            node.destroy()
+        self.simulator.stop()
+
+    def test(self):
+        self.nodes[LEADER].start()
+        self.simulator.go(5)
+        self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
+
+        self.nodes[COMMISSIONER].start()
+        self.simulator.go(5)
+        self.assertEqual(self.nodes[COMMISSIONER].get_state(), 'router')
+
+        # Skip all other Coaps sent by Leader
+        self.skip_coap_messages(COMMISSIONER, '0.02')
+        self.skip_coap_messages(LEADER, '2.04')
+
+        # Commissioner start
+        self.nodes[COMMISSIONER].commissioner_start()
+        self.simulator.go(3)
+        self.skip_coap_messages(COMMISSIONER, '0.02')    # Skip LEAD_PET.req
+
+        # Get CommissionerSesssionId from LEAD_PET.rsp
+        leader_messages = self.simulator.get_messages_sent_by(LEADER)
+        msg = leader_messages.next_coap_message('2.04', assert_enabled=False)
```

Comment:
I found that the uri_path of this message is `None`, maybe because this is a response. So if we add the parameter `/c/lp`, the message will not be found.

Reply:
Sorry. You are right, it is a CoAP ACK response. 
I am fine to flush the message queue and then find '2.04' CoAP message from Leader.

Below is another proposal to find the exact matching rsp.
firstly find LEAD_PET.req from COMMISSIONER, and pass Message ID and Token of the request to next_coap_message() (need to be extended) to find the matching rsp from the LEADER.

Thoughts?